### PR TITLE
docs: remove recommendation for prometheus metric_version=2

### PR DESCRIPTION
**Pull request by reimda**
Remove recommendation for metric_version=2 and add a description of both versions so users can choose which is better for their use case.

### CODE REVIEW

File: plugins/inputs/prometheus/README.md
Comment (powersj) [inline plugins/inputs/prometheus/README.md:134]: Thanks for this. Content looks good - should this be at the top, not buried in here? Or a line in the config to direct users here to learn about the differences?

File: plugins/outputs/prometheus_client/README.md
Comment (powersj) [inline plugins/outputs/prometheus_client/README.md:19]: do we need a similar section about the differences?

File: plugins/inputs/prometheus/README.md
Comment (reimda) [reply plugins/inputs/prometheus/README.md:134]: It can't be closer to the top unless it's before the config section. I think the config section belongs higher up, but a pointer in the config is good idea

File: plugins/outputs/prometheus_client/README.md
Comment (reimda) [reply plugins/outputs/prometheus_client/README.md]: I added a pointer here too

**Review by powersj — APPROVED** [review]

File: plugins/inputs/prometheus/README.md
Comment (srebhan) [inline plugins/inputs/prometheus/README.md:122]: Should we add a comment about round-trip-ability of this metric version versus version 1?

Comment (reimda) [reply plugins/inputs/prometheus/README.md:122]: I didn't write much on round tripping because I didn't have exact details on what works and doesn't work for both versions.  The serializer docs (https://github.com/influxdata/telegraf/blob/master/plugins/serializers/prometheus/README.md) imply that only 2 can round trip, but the input and output only say to make sure you use the same option in input and output if you want to round trip.

I was thinking it would be nice to add unit tests for round tripping if they don't already exist. That way it would be clear exactly what the shortcomings are and we could add clearer docs. I didn't want to hold up this PR on adding comprehensive unit tests though.

Comment (srebhan) [reply plugins/inputs/prometheus/README.md:122]: Fair enough.

**Review by srebhan — APPROVED** [review]
Looks good to me.

**Comment by popey** [comment]
Looks good to me! Thanks @reimda

**Review by popey — APPROVED** [review]

**Comment by wangyuan0916** [comment]
I'm confused... Why it is removed from docs? Isn't metric_version = 1 deprecated ? I saw these sentences from official release document.  https://docs.influxdata.com/telegraf/v1.23/release-notes-changelog/#v113-2019-12-12

>  The Prometheus Format (prometheus) input plugin and Prometheus Client (prometheus_client) output have a new mapping to and from Telegraf metrics, which can be enabled by setting metric_version = 2. **The original mapping is deprecated.**

**Comment by euclidsun** [comment]
And the recommendation for v2 has been removed as well, can anyone share more context why is that?
What is the plan for v1, is it going to be deprecated in next few releases or will it still be around for a long time?

- the v1 deprecation were there for 1.13 ~ 1.17, and removed on 1.18
- the v2 as recommendation were there for 1.13 ~ 1.22
- now it's going to remove the v2 recommendation on 1.23 as well per the above

**Comment by reimda** [comment]
@euclidsun @wangyuan0916 We removed the recommendation for v2 because it produces sparse metrics that don't perform well for some use cases and some storage backends. Since v2 isn't superior in all cases for all users, we didn't want to recommend it in all cases.

We removed the deprecation for v1 because it doesn't produce sparse metrics so it can perform better in some cases. We're not planning to remove it at this point. 

We leave it up to users to choose what version is best for them. Both mappings are in use and at this point we expect to keep both.

**Comment by euclidsun** [comment]
thanks @reimda for the context and background, it helps!

**Pull request merged**
